ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -2459,11 +2730,133 @@ signal_list_t* connect_task_instantiation_and_alias(short PASS, ast_node_t* task
                             /* since the driver net is deleted, copy the spot of the in_net over */
                             input_nets_sc->data[sc_spot_input_old] = (void*)net;
                         } else if ((net != in_net) && (net->combined == false)) {
-                            /* if they haven't been combined already, then join the inputs and output */
+                            long idx_array_size = 0;
+                            long* idx_array = NULL;
+                            long m;
+                            for (m = 0; m < input_nets_sc->size; m++) {
+                                if (input_nets_sc->data[m] == net) {
+                                    idx_array = (long*)vtr::realloc(idx_array, sizeof(long*) * (idx_array_size + 1));
+                                    idx_array[idx_array_size] = m;
+                                    idx_array_size += 1;
+                                }
+                            }
+
                             combine_nets(net, in_net, verilog_netlist);
                             net = NULL;
                             /* since the driver net is deleted, copy the spot of the in_net over */
                             output_nets_sc->data[sc_spot_output] = (void*)in_net;
+
+                            for (m = 0; m < idx_array_size; m++) {
+                                char* main_net_name = input_nets_sc->string[idx_array[m]];
+                                input_nets_sc->data[idx_array[m]] = (void*)in_net;
+                                if ((sc_spot_output = sc_lookup_string(output_nets_sc, main_net_name)) != -1)
+                                    output_nets_sc->data[sc_spot_output] = (void*)in_net;
+                            }
+                            vtr::free(idx_array);
+                        }
+                    }
+                } else {
+                    if (port_size > 1) {
```

Comment:
Long function again: can you divide into helper functions?
This port_size >1 ) ... code seems to exist in a few places (haven't checked if identical, but may be a good candidate for a helper function).

Reply:
For sure, I do appreciate your comments.
Will enhance the code quality and commenting.